### DISCUSSION TITLE
Custom Nameplates with custom colors and the Threat System

### DISCUSSION THREAD

**Discussion by MetalMusicMan** [Backupiseasy/ThreatPlates#381] [Ideas]
Thanks for all of your work on this addon! I have a small idea/request regarding threat coloring and custom nameplate coloring.

Is there any plan to allow the threat system to change colors on custom nameplates that also have a custom color, but ONLY if that nameplate loses threat (as tank) or gains threat (as a dps/healer)?

It would be neat if I had a custom nameplate that was say, Green, but if that nameplate still could change to my designated threat loss color for tanks if I lost threat, as well as my designated threat gain color as a DPS/Healer if I gain threat.

i.e. as a Tank:
- custom nameplate == green
- tank threat nameplate == red
- tank threat LOSS nameplate == pink

In this situation, the custom nameplate that is usually green would turn pink when I lost threat as a tank. It would turn back to green when I once again gained threat. 

i.e. as a Healer/DPS:
- custom nameplate == green
- tank threat nameplate == pink
- dps/healer aggro nameplate == red

In this situation, as a DPS, the nameplate would stay green when the tank had aggro (when it would normally be red), but turn to 
pink if I gained threat as a DPS/Healer.

**Comment by Backupiseasy** [comment]
There is currently no plan to do that, at least not by options in the custom nameplate settings. The case seems to be to special to add that as general options, I think. 

What I am thinking about is adding some kind of customization possibility that is easier than scripting, but still allows to modifiy how Threat Plates shows the nameplate. For example, something like a function that is executed before/after Threat Plates calculates the color/size/transparency/style of a nameplate and - if the function returns a value - uses this value (instead of calculating it itself). If that makes sense :) Not sure if I can explain it well enough. 

I thought about this a scripting (which is also still only have finished) is rather complicated and a lot of custom styles could be achieved a lot easier. 

It still will take a lot of time because first, as the next big update, I first want to remove some core legacy code of TidyPlates. And I am working on that already for 2+ years :)

**Reply by MetalMusicMan** [reply]
Thanks for the reply! I appreciate all your efforts :D
